Tensile/TensileLibLogicToYaml.py

```diff
@@ -0,0 +1,319 @@
+    problemTypeYamlData.append("  MinimumRequiredVersion: {}\n\n".format(versionString['MinimumRequiredVersion']))
+    problemTypeYamlData.append("BenchmarkProblems:\n  -\n    - # ProblemType\n")       
+
+    for problemTypeKey, problemTypeValue in currentIndexProblemType.items():
+     # always print DataType, DestDataType, ComputeDataType, HighPrecisionAccumulate, OperationType fields
+      if problemTypeKey == "DataType" or problemTypeKey == "DestDataType" or problemTypeKey == "ComputeDataType":
+         problemTypeYamlData.append("      {}: {}\n".format(problemTypeKey, DataType(problemTypeValue).toChar().lower()))
+         continue
+      if (problemTypeKey == "HighPrecisionAccumulate" or problemTypeKey == "OperationType" or 
+          problemTypeKey == "TransposeA" or problemTypeKey == "TransposeB" or problemTypeKey == "UseBeta" or problemTypeKey == "Batched"):
+         problemTypeYamlData.append("      {}: {}\n".format(problemTypeKey, problemTypeValue))
+         continue
+      if problemTypeKey in defaultProblemType:
+         if problemTypeValue != defaultProblemType[problemTypeKey]:
+            problemTypeYamlData.append("      {}: {}\n".format(problemTypeKey, problemTypeValue))
+            
+    return ''.join(str(x) for x in problemTypeYamlData)
+            
+def formForkParametersDataForNonMI(currentIndexSolution):
+    forkParametersYamlData = []
+    
+    forkParametersYamlData.append("    - # BenchmarkProblemSizeGroup\n") 
+    forkParametersYamlData.append("      InitialSolutionParameters:\n") 
+    forkParametersYamlData.append("      BenchmarkCommonParameters:\n") 
+    
+    forkParametersYamlData.append("      ForkParameters:\n")
+    macInstruction = ""
+    for forkKey, forkValue in currentIndexSolution.items():
+      if forkKey == "MACInstruction":
+        if forkValue == "MAC": #This is a bug in the library logic file
+          macInstruction ="         - {}: ['MAD']\n".format(forkKey)
+          forkParametersYamlData.append(macInstruction)
+          continue
+        else: 
+          macInstruction ="         - {}: {}\n".format(forkKey,[forkValue])
+          forkParametersYamlData.append(macInstruction)
+          continue
+        
+      if forkKey != "ProblemType" and forkKey != "MatrixInstruction":
+        # Find the matching index for fork key name from list of dictionaries => defaultBenchmarkCommonParameters
+        index =  next((i for i,d in enumerate(defaultBenchmarkCommonParameters) if forkKey in d), None)
+        if index != None:
+          #Strip the leading and trailing [] brackets, trim the leading and trailing spaces
+          DefaultValue = str(defaultBenchmarkCommonParameters[index][forkKey])[1:-1].strip()
+          
+          # remove the "" from the string
+          DefaultValue = re.sub("[\"\']", "", DefaultValue)
+          if str(forkValue) != DefaultValue:
+            if forkKey == "ThreadTile" or forkKey == "WorkGroup":

```

Comment:
Can we make this more generic?
Like, check if the type of forkKey is list or not, or something similar way...
Maybe like this.
  if isInstance(forkValue, list) == False:
    forkValue = [forkValue] # convert to list